OpenRA.Mods.Common/Traits/BotModules/BaseBuilderBotModule.cs

```diff
@@ -194,10 +209,49 @@ void IBotPositionsUpdated.UpdatedDefenseCenter(CPos newLocation)
 
 		void IBotTick.BotTick(IBot bot)
 		{
+			// TODO: this causes pathfinding lag when AI's gets blocked in
 			SetRallyPointsForNewProductionBuildings(bot);
 
-			foreach (var b in builders)
-				b.Tick(bot);
+			BuildingsBeingProduced = new Dictionary<string, int>();
```

Comment:
Use `BuildingsBeingProduced.Clear()` and initialize this once on declaration. We don't want to create a new dict every tick.

Reply:
Yeah, that is clearer.